external/io_netcdf/makefile

```diff
@@ -6,7 +6,7 @@ CODE    = ext_ncd_get_dom_ti.code ext_ncd_get_var_td.code ext_ncd_get_var_ti.cod
 FFLAGS  =  $(FCFLAGS) -I$(NETCDFPATH)/include -I../ioapi_share
 LIBS    = $(LIB_LOCAL) -L$(NETCDFPATH)/lib -lnetcdf
 LIBFFS  = $(LIB_LOCAL) -L$(NETCDFPATH)/lib -lnetcdff -lnetcdf $(NETCDF4_DEP_LIB)
-CPP1    = $(CPP) -P $(TRADFLAG)
+CPP1    = $(CPP) $(TRADFLAG)
```

Comment:
Similarly, would you add a comment as to why we want to remove `-P` from the CPP command from all of the io makefiles.

Reply:
The Makefiles should not hard code options for a specific compiler toolchain if you're going to allow for the possibility that another toolchain is to be used. For the gnu variants, '-P' is already included in the definition of $(CPP), so you're covered. Using the -P option with the PGI compilers means "capture the output in a file instead of sending to stdout", which is not what you want anyway. The options necessary for successful c preprocessor operation need to be included in the $(CPP) variable.